File: src/main/java/compiler/symbol_table/Row.java
package compiler.symbol_table;

public class Row {

    private int id;
    private String type;
    private String name;
    private String value;
    private String methodName;
    private String className;

    public int getId() {
        return id;
    }

    public Row setId(int id) {
        this.id = id;
        return this;
    }

    public String getType() {
        return type;
    }

    public Row setType(String type) {
        this.type = type;
        return this;
    }

    public String getName() {
        return name;
    }

    public Row setName(String name) {
        this.name = name;
        return this;
    }

    public String getValue() {
        return value;
    }

    public Row setValue(String value) {
        this.value = value;
        return this;
    }

    public String getMethodName() {
        return methodName;
    }

    public Row setMethodName(String methodName) {
        this.methodName = methodName;
        return this;
    }

    public String getClassName() {
        return className;
    }

    public Row setClassName(String className) {
        this.className = className;
        return this;
    }

    @Override
    public String toString(){
        return id + " : " + name + " : " + type + " : " + value + " " + methodName + " " + className;
    }

}

File: src/main/scala/compiler/symbol_table/Row.scala
package compiler.symbol_table

class Row {
  private var id: Int = 0
  private var `type`: String = null
  private var name: String = null
  private var value: String = null
  private var methodName: String = null
  private var className: String = null
  private var immutable: Boolean = false

  def getId: Int = id

  def setId(id: Int): Row = {
    this.id = id
    return this
  }

  def getType: String = `type`

  def setType(`type`: String): Row = {
    this.`type` = `type`
    this
  }

  def getName: String = name

  def setName(name: String): Row = {
    this.name = name
    this
  }

  def getValue: String = value

  def setValue(value: String): Row = {
    this.value = value
    this
  }

  def getMethodName: String = methodName

  def setMethodName(methodName: String): Row = {
    this.methodName = methodName
    this
  }

  def getClassName: String = className

  def setClassName(className: String): Row = {
    this.className = className
    this
  }

  def getImmutable: Boolean = immutable

  def setImmutable(immutable : Boolean) : Row = {
    this.immutable = immutable
    this
  }

  override def toString: String = {
    return id + " : " + name + " : " + `type` + " : " + value + " " + methodName + " " + className
  }
}